### PR TITLE
[14.0][FIX] project: Allow access share action from project tasks to project users.

### PR DESCRIPTION
**Pull request by victoralmau**
**Description of the issue/feature this PR addresses**:
Allow access share action from project tasks to project users.

**Steps to reproduce the error**:
- Go to Settings > Users & Companies > Users and create a user only with Project > User permission. (User must not have the Extra Rights > Contact Creation permission).
- Go to Project and go into a task and click on "action".
- Share action is not shown.

**Impacted versions**:
- 14.0
- 15.0
- 16.0
 - master

Ping  @pedrobaeza 

cc @Tecnativa

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by victoralmau** [comment]
Updated description with steps to reproduce the problem.

**Comment by IT-Ideas** [comment]
Hello @mart-e as you wrote the commit that introduced the acl rule you might be the best person to provide a hint here.
Thanks!

File: addons/project/security/project_security.xml [hunk 1]
```diff
@@ -114,5 +114,12 @@
         <field name="groups" eval="[(4, ref('base.group_portal'))]"/>
     </record>
 
+    <record id="access_project_portal_share_rule" model="ir.rule">
+        <field name="name">Project/Task: Allow share tasks</field>
```
Comment (pedrobaeza) [inline addons/project/security/project_security.xml:118]:
```suggestion
        <field name="name">Project/Task: Allow to share tasks</field>
```

File: addons/project/security/project_security.xml [hunk 2]
```diff
@@ -114,5 +114,12 @@
...
+        <field name="model_id" ref="portal.model_portal_share"/>
+        <field name="domain_force">[('res_model', '=', project.task')]</field>
```
Comment (pedrobaeza) [inline addons/project/security/project_security.xml:120]: Is there a global record rule [(1, =, 1)] for the partner_manager group?

**Review by mart-e — COMMENTED** [review]
Before 65530dfd6a0486f44134917f5cc42a2c30b95762 we had no rules, so we had to invent for every transient model. Not all are perfect but I think this one made sense.

This wizard allows to create a portal user to access a record, it makes sense to require "Contact Creation" group to show the option IIMHO

File: addons/project/security/project_security.xml (hunk 2 above)
Comment (victoralmau) [reply addons/project/security/project_security.xml:120]: There are currently no rules.
Ok, I understand what you are saying, with this change, the general [(1, =, 1)] rule should also be added in `portal` addon, right?.

**Comment by victoralmau** [comment]
> Before [65530df](https://github.com/odoo/odoo/commit/65530dfd6a0486f44134917f5cc42a2c30b95762) we had no rules, so we had to invent for every transient model. Not all are perfect but I think this one made sense.
> 
> This wizard allows to create a portal user to access a record, it makes sense to require "Contact Creation" group to show the option IIMHO

I think it is "usual" use case, that a user in odoo does not have the "Contact Creation" permission but only wants to know the link to share task.

**Comment by pedrobaeza** [comment]
@mart-e this wizard is not for inviting a user to be portal user, but to share a task. Let me show you a couple of screenshots:

Selección_008

Selección_009

The alternative may be to have a button "Share" in the project task form that calls with sudo the same wizard.

@victoralmau, yes, if you add a record rule that restricts the seen records, then you should add the [(1, =, 1)] rule to the group that have to see everything. In this case, not adding it will lead to non access to the rest of the "Share" options in other models.

**Comment by mart-e** [comment]
Hi again !
You are right, I was mistaking with the `portal.wizard` model, this one is safer.
However, the issue with this is that it can be an issue in a stable version. If you install `project` after this patch, you won't get the rule from `portal` so this PR won't work.
I am not confident it's a good idea to make this kind of change in 14.0 but it should target the master instead (not sure if the model is still here though)

**Comment by pedrobaeza** [comment]
Then I think is safe enough to not add record rules at all in a stable version.

**Comment by pedrobaeza** [comment]
@mart-e we are thinking about another alternative: simply change the group to "Internal user" instead of "Contact creation". It makes sense, as any internal user doesn't need to create a portal user to share any link with token.

What do you think?

**Comment by C3POdoo** [comment]
Dear @victoralmau,

Thank you for your contribution but the version 14.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**